### PR TITLE
165144378 Authentication Middleware

### PR DESCRIPTION
**Pull request by kossy360**
#### Description
This PR adds authentication middleware interfacing with the authentication module. To be called before all protected routes to perform verification. It attaches the decoded payload to the request object on successful verification

#### Type of change
- Chore
  - add authentication middleware

#### Pivotal Tracker
#165144378

### CODE REVIEW

**Comment by innocentEdosa** [comment]
LGTM

**Comment by Fiyiin** [comment]
LGTM

**Comment by Bobjayafam** [comment]
LGTM

**Review by kayroy247 — COMMENTED** [review]
LGTM

File: src/middlewares/authorization.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import authenticator from '../helpers/authenticator';
+
+const authorization = (req, res, next) => {
+  try {
+    const payload = authenticator.verifyToken(req.get('authorization'));
+    req.user = payload;
```
Comment (profmcdan) [inline src/middlewares/authorization.js]: Remove the extra payload declaration.

**Pull request closed without merging**